src/log.js

```diff
@@ -48,14 +47,84 @@ const _loglevelMap = {
  * @returns {number} The numeric log level
  */
 const numericLogLevel = (name) => {
-  const r = _loglevelMap[name];
+  const r = __loglevelMap[name];
   if (r === undefined) {
     throw new Error(`Not a valid log level: ${name}`);
   }
   return r;
 };
 
-numericLogLevel._loglevelMap = _loglevelMap;
+numericLogLevel.__loglevelMap = __loglevelMap;
+
+/**
+ * Internally helix log passe these messages around.
+ *
+ * Messages are just plain objects with some conventions
+ * regarding their fields:
+ *
+ * @example
+ * ```
+ * const myMessage = {
+ *   // REQUIRED
+ *
+ *   level: 'info',
+ *   timestamp: new Date(),
+ *
+ *   // OPTIONAL
+ *
+ *   // This is what constitutes the actual log message an array
+ *   // of any js objects which are usually later converted to text
+ *   // using `tryInspect()`; we defer this text conversion step so
+ *   // formatters can do more fancy operations (like colorizing certain
+ *   // types; or we could)
+ *   message: ['Print ', 42, ' suckers!']
+ *   exception: {
+ *
+ *     // REQUIRED
+ *     $type: 'MyCustomException',
+ *     name; 'MyCustomException',
+ *     message: 'Some custom exception ocurred',
+ *     stack: '...',
+ *
+ *     // OPTIONAL
+ *     code: 42,
+ *     causedBy: <nested exception>
+ *
+ *     // EXCEPTION MAY CONTAIN ARBITRARY OTHER FIELDS
+ *     ...fields,
+ *   }
+ *
+ *   // MESSAGE MAY CONTAIN ARBITRARY OTHER FIELDS
+ *   ...fields,
+ * }
+ * ```
+ *
+ * @interface Message
+ */
+
+/**
+ * Supplies default values for all the required fields in log messages.
+ *
+ * You may pass this a message that is just a string (as opposed to the
+ * usually required array of message components). The message will then automatically
+ * be wrapped in an array.
+ *
+ * @function
+ * @param {Object} [fields={}] User supplied field values; can overwrite
+ *   any default values
+ * @returns {Message}
+ */
+const makeLogMessage = (fields = {}) => {
+  const r = {
+    level: 'info',
+    timestamp: new Date(),
+    ...fields,
+  };
+  if ('message' in r) {
+    r.message = type(r.message) === String ? [r.message] : r.message;
```

Comment:
wouldn't it be better to check for array instead? eg:

```
Array.isArray(r.message) ? r.message : [r.message]
```

Reply:
Hmm, I don't know. I intentionally chose the more restrictive version in order not to invite any false positives…but I guess it does not matter…